platform/security/realm/security-realm-pki/src/main/java/org/codice/ddf/security/pki/realm/PKIRealm.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.pki.realm;
+
+import ddf.security.SubjectUtils;
+import ddf.security.assertion.Attribute;
+import ddf.security.assertion.AttributeStatement;
+import ddf.security.assertion.SecurityAssertion;
+import ddf.security.assertion.impl.AttributeDefault;
+import ddf.security.assertion.impl.AttributeStatementDefault;
+import ddf.security.assertion.impl.DefaultSecurityAssertionBuilder;
+import ddf.security.claims.Claim;
+import ddf.security.claims.ClaimsCollection;
+import ddf.security.claims.ClaimsHandler;
+import ddf.security.claims.impl.ClaimsParametersImpl;
+import java.security.cert.X509Certificate;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import javax.security.auth.x500.X500Principal;
+import org.apache.shiro.authc.AuthenticationException;
+import org.apache.shiro.authc.AuthenticationInfo;
+import org.apache.shiro.authc.AuthenticationToken;
+import org.apache.shiro.authc.SimpleAuthenticationInfo;
+import org.apache.shiro.realm.AuthenticatingRealm;
+import org.apache.shiro.subject.SimplePrincipalCollection;
+import org.codice.ddf.security.handler.api.AuthenticationTokenType;
+import org.codice.ddf.security.handler.api.BaseAuthenticationToken;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PKIRealm extends AuthenticatingRealm {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PKIRealm.class);
+
+  private List<ClaimsHandler> claimsHandlers = new ArrayList<>();
+
+  private Duration fourHours = Duration.ofHours(4);
```

Comment:
❔ How is the default duration of 4 hours determined? Not sure if it is worth a comment.

Reply:
I just picked something. Honestly, it shouldn't matter at all because we don't care how long these assertions last for, they aren't used for anything and will only last as long as the session lasts. If they expire, we still use them. Only the tokens backed by an IdP really matter as far as expiration goes, which is why the field is there.